ext/bcmath/libbcmath/src/div.c

```diff
@@ -328,6 +305,42 @@ static void bc_do_div(
 	efree(numerator_vectors);
 }
 
+static inline void bc_divide_by_one(bc_num numerator, bc_num *quot, size_t quot_scale)
+{
+	quot_scale = MIN(numerator->n_scale, quot_scale);
+	*quot = bc_new_num_nonzeroed(numerator->n_len, quot_scale);
+	char *qptr = (*quot)->n_value;
+	memcpy(qptr, numerator->n_value, numerator->n_len + quot_scale);
+}
+
+static inline void bc_divide_by_pow_10(
+	const char *numeratorptr, size_t numerator_readable_size, bc_num *quot, size_t quot_size, size_t quot_scale)
+{
+	char *qptr = (*quot)->n_value;
+	if (quot_size <= quot_scale) {
+		/* int is 0 */
+		*qptr++ = 0;
+		for (size_t i = quot_size; i < quot_scale; i++) {
+			*qptr++ = 0;
+		}
+	}
```

Comment:
I think this is better written as follows:

```suggestion
	for (size_t i = quot_size; i <= quot_scale; i++) {
		*qptr++ = 0;
	}
```